### ISSUE TITLE
[sql] How do I create new migrations?

### ISSUE THREAD

**Issue by wesleywil** [tauri-apps/plugins-workspace#196]
So, I created a test table, test it, everything was working but I needed to change the table column, but I couldn't figure out how to do it properly it throws this error in the console  `Uncaught migration 1 was previously applied but is missing in the resolved migrations`,  when I use npm run tauri dev

i tried to do this way
`    .plugin(TauriSql::default().add_migrations(
      "sqlite:main.db",
      vec![Migration{
        version:2,
        description:"alter tables and insert a test",
        sql:include_str!("../migrations/1.sql"),
        kind:MigrationKind::Up,
      }],
    ))`
and this way as well
`    .plugin(TauriSql::default().add_migrations(
      "sqlite:main.db",
      vec![Migration{
        version:1,
        description:"alter tables and insert a test",
        sql:include_str!("../migrations/2.sql"),
        kind:MigrationKind::Up,
      }],
    ))`

I added this in the 1.sql and added the same thing inside 2.sql file as well
` ALTER TABLE movies DROP COLUMN description;

ALTER TABLE movies ADD COLUMN description VARCHAR(300) NOT NULL;

INSERT INTO movies (title, description) VALUES('test title','test description'); `

and even when I go back with the changes, when the app opens always have this error `Uncaught migration 1 was previously applied but has been modified`.

**Comment by aftabnaveed** [comment]
Did existing migration work for you? I tried this but even the initial migration won't work for me


```
.plugin(TauriSql::default().add_migrations(
    "sqlite:planner.db",
    vec![Migration {
      version: 1,
      description: "create planner",
      sql: include_str!("../migrations/1.sql"),
      kind: MigrationKind::Up,
    }],
```

**Comment by wesleywil** [comment]
In my app worked fine, but I created manually the folder and the sql file with the CREATE TABLE... in it, then when I run npm tauri dev everything was working at first, but when I re-run the npm tauri dev was throwing an error, but the database was reading, and I even inserted a new role from the frontend, but I needed to refresh the page  to show the contents of my app, just like you do in a web app, right-clicking and choosing refresh, I didn't test in a build if it would work as well, because when you build there's no refresh.

**Comment by wesleywil** [comment]
I found out what was the problem, I was creating the first migration and as I was creating the table  I saved one of the columns as a text type then I change to varchar all this while the app was running so only caught the first migration, I only found out because I looked inside the DB file using a software to view databases was all my fault in the end. And to create new Migrations is very easy.

First create the SQL file inside the migrations folder like in the example and put the SQL code as you like, for ex: migrations/2.sql

`CREATE  TABLE  books(
    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    title VARCHAR(300),
    image VARCHAR(50),
    description VARCHAR(300),
    status VARCHAR(30),
    rate INTEGER,
    review VARCHAR(300)
);`

inside the main.rs you write something like this.

`    .plugin(TauriSql::default().add_migrations(
      "sqlite:main.db",
      vec![Migration{
        version:1,
        description:"create movies",
        sql:include_str!("../migrations/1.sql"),
        kind:MigrationKind::Up,
      },Migration{
        version:2,
        description:"create books",
        sql:include_str!("../migrations/2.sql"),
        kind:MigrationKind::Up,
      }],
    ))`

That's it, every new migration you create a new file inside the migrations folder and  insert it inside the list in the main.rs with a new version, description and so on.
